api/v1alpha1/instancegroup_types.go

```diff
@@ -648,7 +670,7 @@ func (ig *InstanceGroup) Validate() error {
 			return err
 		}
 
-		if err := config.Validate(spec.Type); err != nil {
+		if err := config.Validate(spec.Type, spec.HasWarmPool()); err != nil {
```

Comment:
Wonder if it makes more sense to add the validations for warm pool to spec validation (and also move the volume validation I added to spec validation, since it checks data from both spec and the config) - we don't seem to have a consistent approach to validation when we need to check data from spec and config - we have cases in both validation functions where we're checking both.

Reply:
This would make sense, we can probably move warm pool validation to `(s *EKSSpec) Validate()` - we need access to both `EKSSpec` and `EKSConfiguration` so we can check if MixedInstancesPolicy or Spot is used.
Same thing for volumes.
I'll look into refactoring this